src/codecs/avif/encoder.rs

```diff
@@ -101,20 +97,19 @@ impl<W: Write> AvifEncoder<W> {
         color: ColorType,
     ) -> ImageResult<()> {
         self.set_color(color);
-        let config = self.config;
+        let encoder = &self.encoder;
         // `ravif` needs strongly typed data so let's convert. We can either use a temporarily
         // owned version in our own buffer or zero-copy if possible by using the input buffer.
         // This requires going through `rgb`.
-        let result = match self.encode_as_img(data, width, height, color)? {
-            RgbColor::Rgb8(buffer) => encode_rgb(buffer, &config).map(|(data, _color_size)| data),
-            RgbColor::Rgba8(buffer) => {
-                encode_rgba(buffer, &config).map(|(data, _color_size, _alpha_size)| data)
-            }
+        let mut fallback = vec![]; // This vector is used if we need to do a color conversion.
+        let result = match Self::encode_as_img(&mut fallback, data, width, height, color)? {
+            RgbColor::Rgb8(buffer) => encoder.encode_rgb(buffer),
```

Comment:
The function call could just be `self.encoder.encode_rgb(buffer)` instead of borrowing `encoder` above.

Reply:
It could. That's a purely style change, since the borrow would happen with the function call anyway, according to the times the borrow checker yelled at me here.